prometheus/client.go

```diff
@@ -33,40 +31,72 @@ func NewClient() (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	client.p8s = p8s
+	client.api = v1.NewAPI(p8s)
```

Comment:
I understand the change, and perhaps with the prometheus client it could make sense as it stateless, but maintain the client could help to close() it or performs another clean operation.
Perhaps for this is ok.
So, no a blocker, just leaving a comment here about the change just in case in the future we may need to review it.

Reply:
I can put it back, np